sql_generators/glean_usage/templates/clients_last_seen_joined.query.sql

```diff
@@ -1,33 +1,46 @@
 WITH baseline AS (
-  SELECT 
+  SELECT
     *
-  FROM 
-    `{{ project_id }}.{{ app_name }}.baseline_clients_last_seen` 
-  WHERE 
+  FROM
+    `{{ project_id }}.{{ app_name }}.baseline_clients_last_seen`
+  WHERE
     submission_date = @submission_date
 ),
 metrics AS (
-  SELECT 
-    * 
-  FROM 
+  SELECT
+    *
+  FROM
     `{{ project_id }}.{{ app_name }}.metrics_clients_last_seen`
   WHERE
     -- The join between baseline and metrics pings is based on submission_date with a 1 day delay,
     -- since metrics pings usually arrive within 1 day after their logical activity period.
     submission_date BETWEEN @submission_date AND DATE_ADD(@submission_date, INTERVAL 1 DAY)
 )
-SELECT 
+SELECT
   baseline.client_id,
   baseline.sample_id,
-  baseline.submission_date, 
+  baseline.submission_date,
   baseline.normalized_channel,
-  * EXCEPT(submission_date, normalized_channel, client_id, sample_id) 
+  * EXCEPT(submission_date, normalized_channel, client_id, sample_id)
 FROM
-  baseline 
+  baseline
 LEFT JOIN metrics
 ON baseline.client_id = metrics.client_id AND
   baseline.sample_id = metrics.sample_id AND
   (
     baseline.normalized_channel = metrics.normalized_channel OR
     (baseline.normalized_channel IS NULL AND metrics.normalized_channel IS NULL)
   )
+{% if app_name == "fenix" -%}
+-- In some rare cases we end up with same client_id in multiple channels in Fenix
+-- In those cases, this union can result in client_id duplicates.
+-- This logic ensures that the resulting table only includes the client from the channel
+-- we've seen first.
+QUALIFY
+  ROW_NUMBER() OVER (
+    PARTITION BY
+      client_id
+    ORDER BY
+      first_seen_date ASC
```

Comment:
I wonder if we should use the client we have seen most recently instead? It seems likely that it would "replace" the client we saw before that.

Reply:
So I'm not sure, because we could end up with rows for the same client_id being mixed over time without a real indicator as to why other than the first_seen_date changing which also could have an effect downstream resulting in duplication when processing clients in batches based on their first_seen_date?

Reply:
I think using the first client seen is better than the last since results per client would be stable over time but it's a rare edge case so I'm not also not sure